protocols/abci/addressing-1174.md

```diff
@@ -0,0 +1,203 @@
+  state and the block that is about to be applied.
+  In other words, for any correct process $p$, and any arbitrary block $u$,
+  if $p$'s CometBFT calls `RequestProcessProposal` on $u$ at height $h$,
+  then $p$'s application's acceptance or rejection **exclusively** depends on $u$ and $s_{p,h-1}$.
+
+* Requirement 5 [`ProcessProposal`, determinism-2]:
+  For any two correct processes *p* and *q*, and any arbitrary
+  block $u$,
+  if $p$'s (resp. $q$'s) CometBFT calls `RequestProcessProposal` on *u* at height $h$,
+  then $p$'s application accepts *u* if and only if $q$'s application accepts $u$.
+  Note that this requirement follows from the previous one and consensus _agreement_.
+
+The requirements expressed above are good enough for most applications using ABCI 1.0 or 2.0.
+They are simple to understand and it is relatively easy to check whether an application's
+implementation of `PrepareProposal` and `ProcessProposal` fulfills them.
+All applications that are able to enforce these properties do not need to reason about
+the internals of the consensus implementation: they can consider it as a black box.
+This is the most desirable situation in terms of modularity between CometBFT and the application.
+
+## Problem Statement
+
+This document is dealing with the case when an application cannot guarantee
+the coherence and/or determinism requirements as stated in the ABCI 1.0 specification.
+
+An example of this is when `ProcessProposal` needs to take inputs from third-party entities
+(e.g. price oracles) that are not guaranteed to provide exactly the same values to
+different processes during the same height.
+Another example is when `ProcessProposal` needs to read the system clock in order to perform its checks
+(e.g. Proposer-Based Timestamp when expressed as an ABCI 1.0 application).
+
+In principle, if an application's implementation of `PrepareProposal` and `ProcessProposal`
+is not able to fulfill coherence and determinism requirements,
+CometBFT cannot guarantee consensus _termination_ in all runs of the system.
+As a result, the application designers must start considering both CometBFT and their application
+as one monolithic block, in order to reason about termination.
+We thus lose the modularity provided when fulfilling the ABCI 1.0 requirements.
+Remember that CometBFT's consensus algorithm (Tendermint) is a well-known algorithm that
+has been studied, reviewed, formally analyzed, model-checked, etc.
+The combination of CometBFT and an arbitrary application as one single algorithm cannot
+leverage that extensive body of research applied to the Tendermint algorithm.
+This situation is risky and undesirable.
+
+So, the questions that arises are the following.
+Can we come up with a set of weaker requirements
+that applications unable to fulfill the current ABCI 1.0 requirements
+can still fulfill?
+Is this set of weaker requirements still strong enough to guarantee consensus _termination_?
+
```

Comment:
```suggestion

> Let's rephrase this a bit at the level of the consensus algorithm. Classically, Tendermint consensus 
> has been analyzed using a function `valid(v, bc(h-1))` that ensured that correct processes agree on
> the evaluation of this function because before starting height `h`, there was agreement on height 
> `h - 1`. We want to relax that by using a function `valid(v, bc(h-1), x_p)` where `x_p` is a local 
> variable at process `p`. As `x_p` may be different from `x_q` for two processes `p` and `w`, 
> using `valid` in `PrepareProposal` 
> and `ProcessProposal` may break Requirements 3 to 5.

```

Reply:
Makes sense. Adding the annotation text for the moment